tests/conftest.py

```diff
@@ -59,3 +72,208 @@ def memory_keyring():
     keyring.set_keyring(MemoryKeyring())
     yield
     keyring.set_keyring(current_keyring)
+
+
+class Platform(models.CraftBaseModel):
+    """Platform definition."""
+
+    build_on: str
+    build_for: str
+
+
+class MyBuildPlanner(models.BuildPlanner):
+    """Build planner definition for tests."""
+
+    def get_build_plan(self) -> list[models.BuildInfo]:
+        arch = util.get_host_architecture()
+        return [
+            models.BuildInfo(
+                "ubuntu-22.04", arch, arch, bases.BaseName("ubuntu", "22.04")
+            ),
+            models.BuildInfo(
+                "ubuntu-24.04", arch, arch, bases.BaseName("ubuntu", "24.04")
+            ),
+        ]
+
+
+@pytest.fixture()
+def features(request) -> dict[str, bool]:
+    """Fixture that controls the enabled features.
+
+    To use it, mark the test with the features that should be enabled. For example:
+
+    @pytest.mark.enable_features("build_secrets")
+    def test_with_build_secrets(...)
+    """
+    features = {}
+
+    for feature_marker in request.node.iter_markers("enable_features"):
+        for feature_name in feature_marker.args:
+            features[feature_name] = True
+
+    return features
+
+
+@pytest.fixture()
+def app_metadata(features) -> craft_application.AppMetadata:
+    with pytest.MonkeyPatch.context() as m:
+        m.setattr(metadata, "version", lambda _: "3.14159")
+        return craft_application.AppMetadata(
+            "testcraft",
+            "A fake app for testing snapcraft",
+            BuildPlannerClass=MyBuildPlanner,
+            source_ignore_patterns=["*.snap", "*.charm", "*.starcraft"],
+            features=craft_application.AppFeatures(**features),
+        )
+
+
+@pytest.fixture()
+def fake_project() -> models.Project:
```

Comment:
we now have a default_project in unit/conftest and a variation of this here

Reply:
I put the two into one as the `fake_project`